src/biokbase/narrative/tests/test_jobmanager.py

```diff
@@ -740,6 +806,219 @@ def test_get_job_info(self):
             infos, {id: ALL_RESPONSE_DATA[MESSAGE_TYPE["INFO"]][id] for id in ALL_JOBS}
         )
 
+    @mock.patch(CLIENTS, get_mock_client)
+    def test_add_errors_to_results__concat_errs__integrated(self):
+        active_ids = [JOB_CREATED, JOB_RUNNING]
+        terminal_ids = [JOB_COMPLETED]
+        job_ids = active_ids + terminal_ids
+
+        check_jobs_err = "Something went wrong in EE2.check_jobs"
+        check_jobs_exc = RuntimeError(check_jobs_err)
+        cancel_job_errs = {
+            job_id: err
+            for job_id, err in zip(
+                job_ids, [f"EE2.check_job err {num}" for num in ["UNO", "DOS"]]
+            )
+        }
+
+        def mock_check_jobs(self, params):
+            raise check_jobs_exc
+
+        def mock_cancel_job(self, job_id):
+            return NarrativeException(
+                None, cancel_job_errs[job_id], None, None, None
+            )
+
+        with mock.patch.object(MockClients, "check_jobs", mock_check_jobs):
+            with mock.patch.object(JobManager, "_cancel_job", mock_cancel_job):
+                output_states = self.jm.cancel_jobs(job_ids)
+
+        exp = {job_id: copy.deepcopy(ALL_RESPONSE_DATA[MESSAGE_TYPE["STATUS"]][job_id]) for job_id in job_ids}
+        for job_id in active_ids:
+            exp[job_id]["error"] = (
+                f"A Job.query_ee2_states error occurred for job with ID {job_id}: {check_jobs_err}"
+                "\n"
+                f"An EE2.cancel_job error occurred for job with ID {job_id}: {cancel_job_errs[job_id]}"
+            )
+
+        self.assertEqual(
+            exp,
+            output_states
+        )
+
+    def test_add_errors_to_results__concat_errs__unit(self):
+        job_ids = ALL_JOBS
+        error_ids = [JOB_RUNNING, JOB_COMPLETED]
+        output_states = get_test_jobs(job_ids)
+
+        check_jobs_err = "Test check_jobs exception"
+        cancel_job_errs = ["Test cancel_job exception UNO", "Test cancel_job exception DOS"]
+
+        self.jm.add_errors_to_results(
+            output_states, error_ids, OutputStateErrMsg.QUERY_EE2_STATES, check_jobs_err
+        )
+        self.jm.add_errors_to_results(
+            output_states, error_ids, OutputStateErrMsg.CANCEL, cancel_job_errs
+        )
+
+        for error_id, cancel_job_err in zip(error_ids, cancel_job_errs):
+            output_state = output_states[error_id]
+            self.assertIn("error", output_state)
+            self.assertEqual(
+                (
+                    f"A Job.query_ee2_states error occurred for job with ID {error_id}: {check_jobs_err}"
+                    "\n"
+                    f"An EE2.cancel_job error occurred for job with ID {error_id}: {cancel_job_err}"
+                ),
+                output_state["error"]
+            )
+
+    def test_add_errors_to_results__cannot_add_err(self):
```

Comment:
Test when trying to add error to existing dict in the results dict in `jm.add_errors_to_results` but there's no corresponding dict there